controllers/subscription/subscription_controller.go

```diff
@@ -246,40 +258,53 @@ func (r *SubscriptionReconciler) HandleUpgrades(ctx context.Context, rhmiSubscri
 		}
 	}
 
-	isInstallPlanDeleted := false
-	currentOperatorVersionName := fmt.Sprintf("%s.v%s", CSVNamePrefix, version.GetVersion())
+	csvNamePrefix := "integreatly-operator"
+	if installation.Spec.Type == string(integreatlyv1alpha1.InstallationTypeManagedApi) {
+		csvNamePrefix = "managed-api-service"
+	}
+
+	currentOperatorVersionName := fmt.Sprintf("%s.v%s", csvNamePrefix, version.GetVersion())
 	if csvFromCatalogSource.Spec.Replaces != currentOperatorVersionName {
+		err = rhmiConfigs.DeleteInstallPlan(ctx, latestRHMIInstallPlan, r.Client)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("error deleting installplan %w", err)
+		}
 
-		if csvFromCatalogSource.Spec.Replaces != latestRHMICSV.Spec.Replaces {
-			err = rhmiConfigs.DeleteInstallPlan(ctx, latestRHMIInstallPlan, r.Client)
-			if err != nil {
-				return ctrl.Result{}, fmt.Errorf("error deleting installplan %w", err)
-			}
+		log.Info("Installplan deleted for the install of patch upgrade")
 
-			isInstallPlanDeleted = true
-			log.Info("Installplan deleted for the install of patch upgrade")
+		err = rhmiConfigs.CreateInstallPlan(ctx, rhmiSubscription, r.Client)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
 
-			err = rhmiConfigs.CreateInstallPlan(ctx, rhmiSubscription, r.Client)
+		err := wait.Poll(time.Second*5, time.Minute*1, func() (done bool, err error) {
+			// gets the subscription with the recreated installplan
+			logrus.Infof("Entered wait.Poll")
```

Comment:
I'd change the log message here to "waiting for the new install plan to be recreated"